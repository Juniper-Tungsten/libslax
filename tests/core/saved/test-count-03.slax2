/* Machine Crafted with Care (tm) by slaxWriter */
version 1.1;

var $x = <message> "Down rev PIC in " _ ../name _ ", " _ name _ ": " _ description;
var $message = <x> _name/text() _ " (" _ id/text() _ ")";
var $count = /in/one _ "-2-" _ /in/three _ "-4-" _ /in/five;
var $sp_one = "uno, Man";
var $sp_two = "dos";
var $sp_three = "tres";
var $sp_four = "quatro";
var $sp_five = "cinco";
var $is = "is";
var $zero = 0;

match / {
    <out> {
        <count> $count;
        var $variable = "this";
        var $test = ($variable ==("this"));
        <test> $test;
        var $test2 = ($variable ==(concat("th", $is)));
        <test> $test2;
        <output> slax:evaluate("$variable == ('th' _ 'is')");
        <test> substring("tent", 1, 3) _ 6 - 10;
        <test> $zero + 10 _ (concat("th", $is, (concat(10, 6))));
        <title> "Translation of " _ name() _ ":";
        apply-templates in/*;
    }
}

match * {
    element name() {
        expr slax:evaluate(concat("$sp_", name()));
    }
}
