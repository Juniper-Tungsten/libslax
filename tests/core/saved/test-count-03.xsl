<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="slax">
  <xsl:variable name="x">
    <message>
      <xsl:value-of select="concat(&quot;Down rev PIC in &quot;, ../name, &quot;, &quot;, name, &quot;: &quot;, description)"/>
    </message>
  </xsl:variable>
  <xsl:variable name="message">
    <x>
      <xsl:value-of select="concat(_name/text(), &quot; (&quot;, id/text(), &quot;)&quot;)"/>
    </x>
  </xsl:variable>
  <xsl:variable name="count" select="concat(/in/one, &quot;-2-&quot;, /in/three, &quot;-4-&quot;, /in/five)"/>
  <xsl:variable name="sp_one" select="&quot;uno, Man&quot;"/>
  <xsl:variable name="sp_two" select="&quot;dos&quot;"/>
  <xsl:variable name="sp_three" select="&quot;tres&quot;"/>
  <xsl:variable name="sp_four" select="&quot;quatro&quot;"/>
  <xsl:variable name="sp_five" select="&quot;cinco&quot;"/>
  <xsl:variable name="is" select="&quot;is&quot;"/>
  <xsl:variable name="zero" select="0"/>
  <xsl:template match="/">
    <out>
      <count>
        <xsl:value-of select="$count"/>
      </count>
      <xsl:variable name="variable" select="&quot;this&quot;"/>
      <xsl:variable name="test" select="($variable =(&quot;this&quot;))"/>
      <test>
        <xsl:value-of select="$test"/>
      </test>
      <xsl:variable name="test2" select="($variable =(concat(&quot;th&quot;, $is)))"/>
      <test>
        <xsl:value-of select="$test2"/>
      </test>
      <output>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:evaluate(&quot;$variable == ('th' _ 'is')&quot;)"/>
      </output>
      <test>
        <xsl:value-of select="concat(substring(&quot;tent&quot;, 1, 3), 6 - 10)"/>
      </test>
      <test>
        <xsl:value-of select="concat($zero + 10, (concat(&quot;th&quot;, $is, (concat(10, 6)))))"/>
      </test>
      <title>
        <xsl:value-of select="concat(&quot;Translation of &quot;, name(), &quot;:&quot;)"/>
      </title>
      <xsl:apply-templates select="in/*"/>
    </out>
  </xsl:template>
  <xsl:template match="*">
    <xsl:element name="{name()}">
      <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:evaluate(concat(&quot;$sp_&quot;, name()))"/>
    </xsl:element>
  </xsl:template>
</xsl:stylesheet>
