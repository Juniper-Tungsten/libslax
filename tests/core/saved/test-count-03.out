<?xml version="1.0"?>
<out>
  <count>1-2-3-4-5</count>
  <test>true</test>
  <test>true</test>
  <output>true</output>
  <test>ten-4</test>
  <test>10this106</test>
  <title>Translation of :</title>
  <one>uno, Man</one>
  <two>dos</two>
  <three>tres</three>
  <four>quatro</four>
  <five>cinco</five>
</out>
